yarn-project/acir-simulator/src/public/executor.ts

```diff
@@ -152,6 +152,12 @@ export class PublicExecutor {
         this.log(`Emitted unencrypted log: "${log.toString('ascii')}"`);
         return Promise.resolve(ZERO_ACVM_FIELD);
       },
+      getPortalContractAddress: async ([aztecAddress]) => {
+        const contractAddress = AztecAddress.fromString(aztecAddress);
+        let portalContactAddress = await this.contractsDb.getPortalContractAddress(contractAddress);
+        if (!portalContactAddress) portalContactAddress = EthAddress.ZERO;
```

Comment:
Thoughts on returning zero here, or throwing. I feel like it should throw

Reply:
i agree -> although can you remind me how the error would bubble up to the user? (what would be their experience)

Reply:
good point, the user should probably have an explicit zero check

Reply:
![image](https://github.com/AztecProtocol/aztec-packages/assets/47148561/22f3e691-7297-4594-b78e-ca140cec7d7d)